File: Leetcode/1001-2000/1208. Get Equal Substrings Within Budget/1208. sliding window.py
class Solution:
    def equalSubstring(self, s: str, t: str, maxCost: int) -> int:
        left, right, sum, ans = 0, 0, 0, 0
        while right < len(s):
            sum += abs(ord(s[right])-ord(t[right]))
            while left <= right and sum> maxCost:
                sum = sum - abs(ord(s[left])-ord(t[left]))
                left += 1
            if (right-left) >= ans:
                ans = 1+(right-left)
            right += 1
        return ans

File: Leetcode/1001-2000/1208. Get Equal Substrings Within Budget/1208 sliding window.java
class Solution {
    public int equalSubstring(String s, String t, int maxCost) {
        int left = 0, right = 0, sum = 0, ans = 0;
        while(right< s.length())
        {
            sum += Math.abs(s.charAt(right) - t.charAt(right));
            while(sum> maxCost)
            {
                sum -= Math.abs(s.charAt(left)-t.charAt(left));
                left++;
            }
            if( right-left>=ans)
                ans = 1+right-left;
            right++;
        }
        return ans;
    }
}
